spring-apps/autosleep-core/src/main/java/org/cloudfoundry/autosleep/ui/web/controller/AutoEnrollmentController.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    @RequestMapping(method = RequestMethod.DELETE, value = "/{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> deleteOrganization(
+            @PathVariable String organizationId) {
+        log.debug("deleteEnrolledOrganization - {}", organizationId);
+        OrgEnrollmentConfig orgConfig = orgEnrollmentRepository.findOne(organizationId);
+        if (orgConfig == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        orgEnrollmentRepository.delete(organizationId);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.PUT, value = "{organizationId}")
+    public ResponseEntity<OrgEnrollmentConfig> enrolOrganization(
+            @PathVariable String organizationId,
+            @RequestBody OrgEnrollmentConfigRequest orgEnrollerConfigRequest, BindingResult result)
+            throws CloudFoundryException, BindException {
+
+        log.debug("enrolOrganization - {}", organizationId);
+        if (!cloudfoundryApi.isValidOrganization(organizationId)) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        validator.validate(orgEnrollerConfigRequest, result);
+        if (result.hasErrors()) {
+            throw new BindException(result);
+        }
+
+        OrgEnrollmentConfig orgEnrollerConfig = OrgEnrollmentConfig.builder()
+                .organizationGuid(organizationId)
+                .idleDuration(orgEnrollerConfigRequest.getIdleDuration() == null ? null
+                        : idleDurationReader
+                                .readParameter(orgEnrollerConfigRequest.getIdleDuration(), false))
+                .excludeSpacesFromAutoEnrollment(
+                        orgEnrollerConfigRequest.getExcludeSpacesFromAutoEnrollment() == null ? null
+                                : excludeFromAutoEnrollmentReader
+                                        .readParameter(
+                                                orgEnrollerConfigRequest
+                                                        .getExcludeSpacesFromAutoEnrollment(),
+                                                false))
+                .autoEnrollment(orgEnrollerConfigRequest.getAutoEnrollment() == null ? null
+                        : autoEnrollmentReader
+                                .readParameter(orgEnrollerConfigRequest.getAutoEnrollment(), false))
+                .state(orgEnrollerConfigRequest.getState() == null ? null
+                        : stateReader.readParameter(orgEnrollerConfigRequest.getState(), false))
+                .build();
+
+        boolean isAlreadyRegistered = orgEnrollmentRepository.exists(organizationId);
+        orgEnrollerConfig = orgEnrollmentRepository.save(orgEnrollerConfig);
+
+        if (isAlreadyRegistered) {
+            return new ResponseEntity<OrgEnrollmentConfig>(orgEnrollerConfig, HttpStatus.OK);
```

Comment:
Good point. Will change this. PATCH makes more sense for altering attributes. I overlooked it.